common/Resources/ti.internal/extensions/util.js

```diff
@@ -0,0 +1,497 @@
+		},
+		isBooleanObject: value => {
+			return typeof value === 'object' && Object.prototype.toString.call(value) === '[object Boolean]';
+		},
+		// isBigIntObject: value => {
+		// 	return Object.prototype.toString.call(value) === '[object BigInt]';
+		// },
+		isSymbolObject: value => {
+			return typeof value === 'object' && Object.prototype.toString.call(value) === '[object Symbol]';
+		},
+		isBoxedPrimitive: function (value) {
+			if (typeof value !== 'object') {
+				return false;
+			}
+			return this.isNumberObject(value)
+				|| this.isStringObject(value)
+				|| this.isBooleanObject(value)
+				// || this.isBigIntObject(value)
+				|| this.isSymbolObject(value);
+		},
+		isNativeError: value => {
+			// if not an instance of an Error, definitely not a native error
+			if (!(value instanceof Error)) {
+				return false;
+			}
+			if (!value || !value.constructor) {
+				return false;
+			}
+			return [ 'Error', 'EvalError', 'RangeError', 'ReferenceError', 'SyntaxError', 'TypeError', 'URIError' ].includes(value.constructor.name);
+
+		},
+		isSet: value => value instanceof Set,
+		isMap: value => value instanceof Map,
+		isDate: value => value instanceof Date,
+		isRegexp: value => value instanceof RegExp || Object.prototype.toString.call(value) === '[object RegExp]'
+	},
+	isArray: value => Array.isArray(value),
+	isBoolean: value => typeof value === 'boolean',
+	isFunction: value => typeof value === 'function',
+	isNull: value => value === null,
+	isNullOrUndefined: value => value === undefined || value === null,
+	isNumber: value => typeof value === 'number',
+	isObject: value => value !== null && typeof value === 'object',
+	isPrimitive: value => (typeof value !== 'object' && typeof value !== 'function') || value === null,
+	isString: value => typeof value === 'string',
+	isSymbol: value => typeof value === 'symbol',
+	isUndefined: value => value === undefined,
+	log: string => {
+		const date = new Date();
+		const time = `${date.getHours().toString().padStart(2, '0')}:${date.getMinutes().toString().padStart(2, '0')}:${date.getSeconds().toString().padStart(2, '0')}`;
+		// Produces output like: "21 Feb 10:04:23 - message"
+		console.log(`${date.getDate()} ${MONTHS[date.getMonth()]} ${time} - ${string}`);
+	},
+	print: (...args) => console.log(args.join('')), // FIXME: Shouldn't add trailing newline like console.log does!
+	puts: (...args) => console.log(args.join('\n')),
+	error: (...args) => console.error(args.join('\n')),
+	debug: string => console.error(`DEBUG: ${string}`)
+};
+
+util.isBuffer = () => false; // FIXME: Check for Ti.Buffer? for node/browserify buffer?
+util.isDate = value => util.types.isDate(value);
+util.isError = value => util.types.isNativeError(value);
+util.isRegexp = value => util.types.isRegexp(value);
+
+// FIXME: Our String.format is not very forgiving. It sort-of is supposed to do the same thing, but blows up easily
+// util.format = String.format;
```

Comment:
Can this be removed? since you wrote your own format function further down?

Reply:
yeah, it was a reminder as I tried to use our String.format as a placeholder and it broke right away.